src/main/java/com/thealgorithms/maths/ComplexNumberUtil.java

```diff
@@ -0,0 +1,267 @@
+package com.thealgorithms.maths;
+
+public class ComplexNumberUtil {
+
+    public static class ComplexNumber {
+        public final double REAL;
+        public final double IMAGINARY;
+
+        public ComplexNumber(double real, double imaginary) {
+            REAL = real;
+            IMAGINARY = imaginary;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj instanceof ComplexNumber num) {
+                return this.REAL == num.REAL && this.IMAGINARY == num.IMAGINARY;
+            }
+
+            return super.equals(obj);
+        }
+
+        @Override
+        public String toString() {
+            return REAL + ((IMAGINARY < 0) ? (" - " + Math.abs(IMAGINARY)) : (" + " + IMAGINARY)) + "i";
+        }
+    }
+
+    public final static ComplexNumber ZERO = new ComplexNumber(0, 0);
+    public final static ComplexNumber ONE = new ComplexNumber(1, 0);
+    public final static ComplexNumber TWO = new ComplexNumber(2, 0);
+    public final static ComplexNumber PLUS_I = new ComplexNumber(0, 1);
+    public final static ComplexNumber MINUS_I = new ComplexNumber(0, -1);
+
+    /**
+     * add two complex numbers
+     * @param num1 the first complex number
+     * @param num2 the second complex number
+     * @return the sum of num1 and num2
+     */
+    public static ComplexNumber add(ComplexNumber num1, ComplexNumber num2) {
+        return new ComplexNumber(num1.REAL + num2.REAL, num1.IMAGINARY + num2.IMAGINARY);
+    }
```

Comment:
Why did you decide to make it a `static` method? In the _standard_ implementation of [`Complex`](https://commons.apache.org/proper/commons-math/javadocs/api-3.6.1/org/apache/commons/math3/complex/Complex.html#add(org.apache.commons.math3.complex.Complex)) `add` is an _object_ method.

Reply:
I decided to make it a static method for a few reasons:

<b>Uniformity in Code</b>: The file [MatrixUtil.java](https://github.com/TheAlgorithms/Java/blob/master/src/main/java/com/thealgorithms/maths/MatrixUtil.java), which is similar to this, uses static methods for the operations.
<b>Ease of Expansion</b>: Creating independent static methods makes it easier to add more advanced operations for complex numbers than integrating additional code into the ComplexNumber class itself.
<b>Clarity</b>: I find it easier to conceptualize operations as direct interactions between two entities, A and B (i.e., how do you add them), rather than how one adds B to A.

These are my reasons, but if you believe it should be an instance method, I would be happy to make that change.

Reply:
Making them `static` you loose all of the _object oriented magic_.  The whole thing could be then simpler, because the `ComplexNumberUtil` would not be needed. Furthermore, you have marked both of the fields `REAL` and `IMAGINARY` as `final`, so it is pretty clear that the `add` method will not change them.